interop/xds/server/server.go

```diff
@@ -78,8 +83,47 @@ func (s *testServiceImpl) EmptyCall(ctx context.Context, _ *testpb.Empty) (*test
 }
 
 func (s *testServiceImpl) UnaryCall(ctx context.Context, in *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
+	response := &testpb.SimpleResponse{ServerId: s.serverID}
+
+	for headerValue := range getRPCBehaviorMetadata(ctx) {
+		if rpcStatus := getStatusForRPCBehaviorMetadata(headerValue); rpcStatus != nil {
+			return response, status.Err(rpcStatus.Code(), rpcStatus.Message())
+		}
+	}
+
+	response.Hostname = s.hostname
 	grpc.SetHeader(ctx, metadata.Pairs("hostname", s.hostname))
-	return &testpb.SimpleResponse{ServerId: s.serverID, Hostname: s.hostname}, nil
+	return response, status.Err(codes.OK, "")
+}
+
+func getRPCBehaviorMetadata(ctx context.Context) map[string]bool {
+	var mdRPCBehavior []string
+	if md, ok := metadata.FromIncomingContext(ctx); !ok {
+		logger.Error("failed to receive metadata")
+		return nil
+	} else {
+		mdRPCBehavior = md.Get(xdsinterop.RPCBehaviorMetadataKey)
+	}
+
+	rpcBehaviorMetadata := make(map[string]bool)
```

Comment:
I just realized this about the C++ implementation (I commented on theirs too) and this, this needs to be a slice. Using a map takes away the ordering of the comma separated values, which breaks this line from the gRFC: "The `rpc-behavior` header value can have multiple options separated by commas. In that case, the value should be split by commas and the options ***should be applied in the order specified***" see the Java implementation for deterministic list ordering: https://github.com/grpc/grpc-java/blob/42b4c61d5e869b1534808204bc71df84720f5087/interop-testing/src/main/java/io/grpc/testing/integration/XdsTestServer.java#L403.